src/engine/execute.js

```diff
@@ -101,24 +105,20 @@ var execute = function (sequencer, thread) {
     // it's treated as a predicate; if not, execution will proceed as a no-op.
     // For single-field shadows: If the block has a single field, and no inputs,
     // immediately return the value of the field.
-    if (!blockFunction) {
+    if (typeof blockFunction === 'undefined') {
         if (isHat) {
             // Skip through the block (hat with no predicate).
             return;
+        }
+        var keys = Object.keys(fields);
+        if (keys.length === 1 && Object.keys(inputs).length === 0) {
+            // One field and no inputs - treat as arg.
+            handleReport(fields[keys[0]].value);
         } else {
-            if (Object.keys(fields).length === 1 &&
-                Object.keys(inputs).length === 0) {
-                // One field and no inputs - treat as arg.
-                for (var fieldKey in fields) { // One iteration.
```

Comment:
A little note here... There was a merge conflict at this point which I had to resolve... We've lost a 'hasOwnProperty' call in the resolution which I don't believe is an issue, but I just wanted to raise it.